src/algorithms/rsa.rs

```diff
@@ -11,6 +11,52 @@ use zeroize::{Zeroize, Zeroizing};
 use crate::errors::{Error, Result};
 use crate::traits::{PrivateKeyParts, PublicKeyParts};
 
+// The number of 32-bit words per element in the risc0 RSA syscalls
+// Must match risc0_zkvm_platform::syscall::rsa::WIDTH_WORDS
+#[cfg(all(target_os = "zkvm", target_arch = "riscv32"))]
+const WIDTH_WORDS: usize = 96;
+
+#[cfg(all(target_os = "zkvm", target_arch = "riscv32"))]
+extern "C" {
+    fn modpow_65537(
+        recv_buf: *mut [u32; WIDTH_WORDS],
+        in_base: *const [u32; WIDTH_WORDS],
+        in_modulus: *const [u32; WIDTH_WORDS],
+    );
+}
+
+#[cfg(all(target_os = "zkvm", target_arch = "riscv32"))]
+fn risc0_modpow_65537(base: &BigUint, modulus: &BigUint) -> BigUint {
+    // Ensure inputs fill an even number of words
+    let mut base = base.to_bytes_le();
+    if base.len() % 4 != 0 {
+        base.resize(base.len() + (4 - (base.len() % 4)), 0);
+    }
+    let mut modulus = modulus.to_bytes_le();
+    if modulus.len() % 4 != 0 {
+        modulus.resize(modulus.len() + (4 - (modulus.len() % 4)), 0);
+    }
+    let base: [u32; WIDTH_WORDS] = base.chunks(4)
+        .map(|word| u32::from_le_bytes(word.try_into().unwrap()))
+        .collect::<Vec<u32>>()
+        .try_into()
+        .unwrap();
+    let modulus: [u32; WIDTH_WORDS] = modulus.chunks(4)
+        .map(|word| u32::from_le_bytes(word.try_into().unwrap()))
+        .collect::<Vec<u32>>()
+        .try_into()
+        .unwrap();
```

Comment:
```suggestion
    let base = base.to_u32_digits();
    let modulus = modulus.to_u32_digits();
```

Reply:
`to_u32_digits` does not exist in `num-bigint-dig`, it's only in `num-bigint`